### PR TITLE
W-14153679-idempotent-message-validator

### PR DESCRIPTION
**Pull request by Cristian-Venticinque**

### CODE REVIEW

File: modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+//end::component-title[]
+//
+
+//
+//tag::component-short-description[]
+//     Short description of the form "Do something..." 
+//     Example: "Configure log messages anywhere in a flow."
+
+Ensures that only unique messages continue through a flow execution by checking the unique ID of the incoming message.
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc]:
```suggestion
Ensure that only unique messages continue through a flow execution by checking the unique ID of the incoming message.
```

File: modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+//
+
+
+//SECTION: EXAMPLES
+//
+//tag::component-examples-title[]
+
+== Examples
+
+The following examples show how to configure an Idempotent Message Validator to extract an ID from an HTTP request to use as a the unique identifier and how to hash the payload using a DataWeave expression and use the message ID as the unique identifier.
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc]:
```suggestion
The examples show how to configure an Idempotent Message Validator to extract an ID from an HTTP request for use as a unique identifier and how to hash the payload using a DataWeave expression and use the message ID as the unique identifier. 
```

File: modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+//end::component-examples-title[]
+//
+//
+//tag::component-xml-ex1[]
+[[example1]]
+
+=== Example: Generate an ID from an HTTP Request
+
+The following example shows an Idempotent Message Validator configured to extract the query parameter id from the HTTP request and use that value as a unique identifier to filter the message:
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc]:
```suggestion
The following example shows an Idempotent Message Validator configured to extract the query parameter ID from the HTTP request and use that value as a unique identifier to filter the message:
```

File: modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+//end::component-xml-ex1[]
+//
+//
+//tag::component-xml-ex2[]
+[[example2]]
+
+=== Example: Generate an ID by Hashing the Payload With DataWeave
+
+The following example shows an Idempotent Message Validator configured to hash the payload using a DataWeave expression, store the result in a persistent object store, and then use the processed message IDs as the unique identifier to filter the message:
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc]:
```suggestion
The following example shows an Idempotent Message Validator configured to hash the payload using a DataWeave expression, store the result in a persistent object store, and then use the processed message IDs as unique identifiers for filtering the message:
```

File: modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+//end::component-xml-ex1[]
+//
+//
+//tag::component-xml-ex2[]
+[[example2]]
+
+=== Example: Generate an ID by Hashing the Payload With DataWeave
+
+The following example shows an Idempotent Message Validator configured to hash the payload using a DataWeave expression, store the result in a persistent object store, and then use the processed message IDs as unique identifiers for filtering the message:
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc:169]: don't see the filtering part

File: modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+----
+<flow name="myFlow">
+  <http:listener doc:name="Listener" config-ref="HTTP_Listener_config" path="/"/>
+  ...
+  <idempotent-message-validator doc:name="Idempotent Message Validator" idExpression="
+    #[%dw 2.0
+      import dw::Crypto
+      output application/octet-stream
+      ---
+      Crypto::hashWith(payload,'MD5')]">
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc:181]: check this example; see the location of the closing square bracket

File: modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+
+//end::component-title[]
+//
+
+//
+//tag::component-short-description[]
+//     Short description of the form "Do something..." 
+//     Example: "Configure log messages anywhere in a flow."
+
+Ensure that only unique messages continue through a flow execution by checking the unique ID of the incoming message.
```
Comment (dukesphere) [inline modules/ROOT/pages/_partials/mule-components/4.4/component-idempotent-message-validator.adoc:14]: check this description; component just seems to be establishing a unique identifier for a message but filtering could be one use case

**Review by dukesphere — APPROVED** [review]

**Review by dukesphere — APPROVED** [review]

**Review by Crispy-Salesforce — APPROVED** [review]

**Pull request merged**